jbmc/src/java_bytecode/README.md

```diff
@@ -25,7 +25,80 @@ To be documented.
 
 \subsection java-class-section How a java program / class is represented in a .class
 
-To be documented.
+Every Java class is compiled into a .class file, inner classes, anonymous
+classes or classes for tableswitches are compiled into their own .class
+files.
+
+There exists an [official
+specification](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html)
+
+Each class files contains information about its version, the constant pool,
+information about the contained class, its super-class, its implemented
+interfaces, its fields, methods and finally additional attributes.
+
+The content of a .class file can be inspected via the `javap` tool which is part
+of the JDK installation.
+
+In general, all variable length entries in a class file are represented by first
+having an integer `n` that specified the number of entries and then an array of
+`n` such entries in the file.
+
+The integer format used in class files are unsigned integers of size 8/16/32
+bit, which are named as `u1`/`u2`/`u4`.
+
+\subsection Access flags
+
+The JVM specification defines different access flags, e.g., `final`, `static`,
+`protected`, `private` etc. where different ones are applicable to the class
+itself, its fields or methods. All access flags are represented as bits, the set
```

Comment:
This sentence is not terribly clear to me. Consider something like "All access flags are represented as constant u2 values whose flag names are prefixed with`ACC_`, for example, ACC_PUBLIC is 0x0001.